src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/AccessorBuilder.cs

```diff
@@ -152,40 +150,24 @@ private static object Make<T>() where T : new()
 
             if (typeof(DeclaringType).IsValueType)
             {
-                var setMethod = propInfo.SetMethod.CreateDelegate<StructSetDelegate<DeclaringType, PropertyType>>();
+                var setMethod = propInfo.SetMethod!.CreateDelegate<StructSetDelegate<DeclaringType, PropertyType>>();
 
-                return (ref object obj, object val) =>
+                return (ref object obj, object? val) =>
                 {
                     var unboxed = (DeclaringType)obj;
-                    setMethod(ref unboxed, (PropertyType)val);
-                    obj = unboxed;
+                    setMethod(ref unboxed, (PropertyType)val!);
+                    obj = unboxed!;
                 };
             }
             else
             {
-                var setMethod = propInfo.SetMethod.CreateDelegate<Action<DeclaringType, PropertyType>>();
+                var setMethod = propInfo.SetMethod!.CreateDelegate<Action<DeclaringType, PropertyType>>();
 
-                return (ref object obj, object val) =>
+                return (ref object obj, object? val) =>
                 {
-                    setMethod((DeclaringType)obj, (PropertyType)val);
+                    setMethod((DeclaringType)obj, (PropertyType)val!);
                 };
             }
         }
     }
-
-    internal static class CreateDelegateExtension
```

Comment:
I'd prefer we haven't removed dead code as part of this PR.. Are you sure it's not called by reflection?

Reply:
This code used to be called in this file above - however, when we introduced https://github.com/dotnet/runtime/pull/36680 in 5.0, this became dead code that was never called (because the method directly on MethodInfo takes precedence over extension methods).

I'll add this dead code back and nullable annotate it. But we should remove it sometime.